src/containers/other/PriceWidget/PriceWidget.tsx

```diff
@@ -0,0 +1,206 @@
+			setData({
+				tokenPrice: wildPriceUsd,
+				percentageChange: wildPercentageChange,
+				tokenTitle: LABELS.WILD_TICKER,
+				urlList: WildUrlRowData,
+				urlCMC: URLS.COIN_MARKET_CAP_WILD,
+				dividerCopy: LABELS.DIVIDER_COPY_WILD,
+			});
+		}
+	}, [
+		isRoot,
+		wildPercentageChange,
+		wildPriceUsd,
+		zeroPercentageChange,
+		zeroPriceUsd,
+	]);
+
+	useEffect(() => {
+		isMounted.current = true;
+		getTokenData();
+		return () => {
+			isMounted.current = false;
+		};
+	}, [getTokenData]);
+
+	/////////////////////
+	// React Fragments //
+	/////////////////////
+
+	//- Current Price
+	const currentPriceDetails = (size: string, title?: string) => {
+		const isSizeSmall =
+			size === Size.SML ? styles.CurrentPriceSml : styles.CurrentPriceLrg;
+
+		return (
+			<div className={size === Size.SML ? styles.Details : ''}>
+				{title && (
+					<span
+						data-testid={selectors.tokenTitle}
+						className={styles.TickerName}
+					>
+						{title}
+					</span>
+				)}
+				<div className={styles.CurrentPriceContainer}>
+					<span className={isSizeSmall}>
+						${formatByDecimalPlace(Number(data?.tokenPrice), 4)}
+					</span>
+					<span
+						className={`${styles.PercentageChange} ${
+							Number(data?.percentageChange) < 0
+								? styles.PercentageChangeNegative
+								: styles.PercentageChangePositive
+						}`}
+					>
+						{size === Size.LRG
+							? `(${formatByDecimalPlace(Number(data?.percentageChange), 1)}%)`
+							: `${formatByDecimalPlace(Number(data?.percentageChange), 1)}%`}
+					</span>
+				</div>
+			</div>
+		);
+	};
+
+	//- External Url
+	const externalUrl = () => (
+		<div className={styles.ExternalUrl}>
+			<a
+				data-testid={selectors.coinMarketCapRedirect}
+				href={data?.urlCMC}
+				target="_blank"
+				rel="noreferrer"
+			>
+				{LABELS.COIN_MARKET_CAP_BTN}
+			</a>
+			<img alt={LABELS.EXTERNAL_URL_ICON} src={openExternalUrl} />
+		</div>
+	);
+
+	//- Divider
+	const divider = () => (
+		<>
+			<p data-testid={selectors.dividerCopy} className={styles.DividerCopy}>
+				{data?.dividerCopy}
+			</p>
+			<div className={styles.Divider}></div>
+		</>
+	);
+
+	//- Url Icon Row
+	const externalUrlRow = (
+		urlList?: {
+			title: string;
+			href: string;
+			src: string;
+		}[],
+	) => (
+		<div className={styles.UrlRowContainer}>
+			{urlList?.map((item) => (
+				<a key={item.title} target="_blank" rel="noreferrer" href={item.href}>
+					<div className={styles.IconWrapper}>
+						<img alt={`${item.title}-logo`} src={item.src} />
+						<p>{item.title}</p>
+					</div>
+				</a>
+			))}
+		</div>
+	);
+
+	const dropdownButton = (
+		<div
+			data-testid={selectors.dropDownButton}
+			className={styles.PriceContainer}
+		>
+			{currentPriceDetails(Size.SML, data?.tokenTitle)}
+		</div>
+	);
+
+	const dropdownContent = (
+		<div
+			data-testid={selectors.dropDownContent}
+			className={`${styles.DropdownContent}`}
+		>
+			<div className={`${styles.Section} ${styles.SectionBaseline}`}>
```

Comment:
nit: you can use the `classnames` library for combined class names, e.g.:

```jsx
import classNames from 'classnames';
...
<div className={classNames(styles.Section, styles.SectionBaseLine)}></div>
...
```

You can also handle conditional classes like:

```jsx
import classNames from 'classnames/bind';
const cx = classNames.bind(styles);
...
<div className={cx(styles.Container, { Visible: isVisible })></div>
...
```

Not necessary for approval though!